### ISSUE TITLE
mpg as Length⁻²

### ISSUE THREAD

**Issue by Erutuon** [sharkdp/numbat#271]
In numbat/modules/units/us_customary.nbt, `mpg` is listed as `Length / Volume`, but in the CLI it shows as `Length⁻²`.

```
>>> mpg

  mpg

      = 1 mpg    [Length⁻²]>>> mpg
>>> mile / gallon

  mile / gallon

    = 1.10112e+12 mi⁻²    [Length⁻²]
>>> 1 mpg -> mi^-2

  1 mpg ➞ mile^(-2)

    = 1.10112e+12 mi⁻²    [Length⁻²]
```

I guess this is because `Volume` is `Length^3` in numbat/modules/core/dimensions.nbt, so `Length` divided by `Volume` is `Length / Volume^3`, which is `Length^-2`. Mathematically correct, but pretty bizarre and I guess not correct. Not sure how to solve this, but if the gallons in `mpg` had a different dimension as an amount of substance rather than an extent of three-dimensional space, this weird dimension resolution wouldn't happen.

**Comment by mhmdanas** [comment]
Somewhat relevant: https://github.com/sharkdp/insect/issues/343

**Comment by sharkdp** [comment]
@Erutuon any update on this?

**Comment by Erutuon** [comment]
Apologies for the lack of response. The XKCD article does make sense of what the simplification means, and I guess it gives insight into some problems, like how to calculate how fast on average the gas level in a tank goes down per distance traveled.

I was thinking about this and imagined a feature where you can declare the `Length^3` part of the dimension immutable or indivisible, so that it can't be simplified with the distance traveled, but I don't know how that would work syntactically or where it would have practical benefit, because unit conversion works fine. Defining a `FuelLength` has that effect, though for converting between miles per gallon and km per gallon and liters per 100 km, you'd need `fuel_gallon` and `fuel_liter` (as well as `hectokilometer`, though that's unrelated 😊).

In summary like you say it's initially counterintuitive, but that is how it is when you learn something for the first time.
